bug fixes and init pagination

Melkiades opened the pull request:
fixes #929 still some things to brew

Melkiades commented on tests/testthat/test-exporters.R:
```diff
@@ -260,9 +260,6 @@ test_that("Can create flextable object that works with different styles", {
   ft <- tt_to_flextable(tbl, total_width = 20)
   expect_equal(sum(unlist(nrow(ft))), 20)
 
-  ft2 <- tt_to_flextable(tbl, paginate = TRUE, lpp = 20, verbose = TRUE)
-  expect_equal(length(ft2), 2)
```
pagination to come

Melkiades commented on tests/testthat/test-exporters.R:
```diff
@@ -312,7 +308,7 @@ test_that("Can create flextable object that works with different styles", {
   expect_equal(flextable::nrow_part(topleft_t2, part = "header"), 2L)
   expect_equal(flextable::nrow_part(topleft_t1a, part = "header"), 1L)
-  expect_equal(flextable::nrow_part(topleft_t1b, part = "header"), 2L)
+  expect_equal(flextable::nrow_part(topleft_t1b, part = "header"), 1L)
```
this is the right answer because it still stays in the same cell

shajoezhu commented on NEWS.md:
```diff
@@ -4,17 +4,26 @@
 
 ### Bug Fixes
  * Fixed `"\n"` newline issues in `as_html` by relying onto output devices for newline handling. Added `expand_newlines = FALSE` default to allow previous behavior.
  * `keep_split_levels` throws now an error if the user requests to keep levels that are not present in data.
```
```suggestion
 * `keep_split_levels()` throws now an error if the user requests to keep levels that are not present in data.
```

shajoezhu commented on NEWS.md:
```diff
@@ -4,17 +4,26 @@
 ### Bug Fixes
  * Fixed `"\n"` newline issues in `as_html` by relying onto output devices for newline handling. Added `expand_newlines = FALSE` default to allow previous behavior.
  * `keep_split_levels` throws now an error if the user requests to keep levels that are not present in data.
+ * Fixed issue with removal of horizontal lines in `tt_as_flextable` header when title was added.
```
```suggestion
 * Fixed issue with removal of horizontal lines in `tt_as_flextable()` header when title was added.
```

shajoezhu commented on tests/testthat/test-exporters.R:
```diff
@@ -361,3 +357,67 @@ test_that("export_as_doc works thanks to tt_to_flextable", {
+  main_footer(tbl) <- c("Some Footer", "Mehr")
+  prov_footer(tbl) <- "Some prov Footer"
+
+  expect_silent(out <- tt_to_flextable(tbl, paginate = TRUE, lpp = 100))
```
can we snapshot this?

shajoezhu replied:
The reason I ask is that I was thinking to write a s3 function of as_flextable.rtables, then we can fully utilize flextable functions such as information_data_chunk, and etc, instead of us writing functions. perhaps next version. perhaps in the next version of this function, we could do something like this.

Melkiades replied:
I see extending as_flextable... It may makes sense but we need probably to import it at that time? 
I can snapshot that yes, but only the data, the style is a bit more complex

Melkiades replied:
I will work on the pagination precisely in the next PR

shajoezhu commented on R/tt_export.R:
```diff
@@ -631,7 +631,12 @@ export_as_docx <- function(tt,
     if (isFALSE(titles_as_header) || isTRUE(footers_as_text)) {
       # Ugly but I could not find a getter for font.size
       font_sz <- flex_tbl$header$styles$text$font.size$data[1, 1]
```
change font_sz to font_sz_body, keep aligned with font_sz_footer

shajoezhu replied:
or consider, adding officer_font_sz_footer, as it is used in officer

shajoezhu commented on R/tt_export.R:
```diff
@@ -631,7 +631,12 @@ export_as_docx <- function(tt,
       # Ugly but I could not find a getter for font.size
       font_sz <- flex_tbl$header$styles$text$font.size$data[1, 1]
-      font_sz_footer <- flex_tbl$header$styles$text$font.size$data[1, 1] - 1
+      footer_sizes <- flex_tbl$footer$styles$text$font.size$data
```
footer_sizes and footer_sizes are bit confusing with the naming.

Melkiades commented:
@shajoezhu I was getting a bit lost between functions, so I split the flextable and data.frame transformers into other files. No other changes since last review (beside the comments' fixes). I think we can merge it and then I will continue with pagination fixes

shajoezhu reviewed (approved):
lgtm! thanks @Melkiades
